File: JavaScript/8js 2/hotpo.js
/*
|--------------------------------------------------------------------------
| Collatz Conjecture (3n+1)
|--------------------------------------------------------------------------
|
*/

function hotpo(n) {
   return n === 1 ? 0 : 1 + hotpo(n % 2 ? n * 3 + 1 : n / 2);
}


const hotpo = (n) => (n < 2 ? 0 : hotpo(n % 2 ? 3 * n + 1 : n / 2) + 1);


function hotpo(n) {
   let res = 0;
   while (n > 1) {
      res++;
      n = n % 2 ? n * 3 + 1 : n / 2;
   }
   return res;
}


console.log(hotpo(1), 0);
console.log(hotpo(5), 5);
console.log(hotpo(6), 8);
console.log(hotpo(23), 15);

File: Python/8py 2/hotpo.py
"""
|--------------------------------------------------------------------------
| Collatz Conjecture (3n+1)
|--------------------------------------------------------------------------
|
"""

def hotpo(n):
   return 0 if n == 1 else 1 + hotpo(n * 3 + 1 if n % 2 else n / 2)


hotpo = f = lambda n:n>1and-~f([n/2,3*n+1][n%2])


def hotpo(n):
   res = 0
   while n != 1:
      n = 3 * n + 1 if n % 2 else n / 2
      res += 1
   return res


print(hotpo(1), 0)
print(hotpo(5), 5)
print(hotpo(6), 8)
print(hotpo(23), 15)